File: app/src/main/java/com/mad/pogoenhancer/Logger.java
package com.mad.pogoenhancer;

import android.util.Log;

import timber.log.Timber;


public class Logger {
    public static void fatal(String TAG, String message) {
        Timber.wtf(message);
        Log.wtf(TAG, message);
    }

    public static void error(String TAG, String message) {
        Log.e(TAG, message);

    }

    public static void warning(String TAG, String message) {
        Log.w(TAG, message);

    }

    public static void info(String TAG, String message) {
        Log.i(TAG, message);
    }

    public static void debug(String TAG, String message) {
        if (BackendStorage.getInstance().isExLog()) {
            Log.d(TAG, message);
        }
    }

    public static void pdebug(String TAG, String message) {
        Log.d(TAG, message);
    }
}

File: app/src/main/cpp/proto_hook/Logger.cpp
#include <android/log.h>
#include "Logger.h"
#include "ProtoCache.h"

#define LOG_TAG "ProtoHookC"
#define LOGD(...) __android_log_print(ANDROID_LOG_DEBUG,LOG_TAG,__VA_ARGS__)
#define LOGI(...) __android_log_print(ANDROID_LOG_INFO,LOG_TAG,__VA_ARGS__)
#define LOGW(...) __android_log_print(ANDROID_LOG_WARN,LOG_TAG,__VA_ARGS__)
#define LOGE(...) __android_log_print(ANDROID_LOG_ERROR,LOG_TAG,__VA_ARGS__)
#define LOGF(...) __android_log_print(ANDROID_LOG_FATAL,LOG_TAG,__VA_ARGS__)
#define LOGS(...) __android_log_print(ANDROID_LOG_SILENT,LOG_TAG,__VA_ARGS__)
#define LOGV(...) __android_log_print(ANDROID_LOG_VERBOSE,LOG_TAG,__VA_ARGS__)

void Logger::fatal(const string& message) {
    LOGF("%s", message.c_str());
}

void Logger::error(const string& message) {
    LOGE("%s", message.c_str());
}

void Logger::warning(const string& message) {
    LOGW("%s", message.c_str());
}

void Logger::info(const string& message) {
    LOGI("%s", message.c_str());
}

void Logger::pdebug(const string& message) {
    LOGD("%s", message.c_str());
}

void Logger::debug(const string& message) {
    LOGD("%s", message.c_str());
}
